File: LearnAndroid/design/src/main/java/com/showdy/hotfix/MainActivity.java
package com.showdy.hotfix;

import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.showdy.interceptor.R;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.btn_test).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                WrongUtils.test();
            }
        });
    }
}

File: LearnAndroid/design/src/main/java/com/showdy/interceptor/MainActivity.kt
package com.showdy.interceptor

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.showdy.interceptor.interceptor.ReceivedServiceInterceptor

@Suppress("UNREACHABLE_CODE")
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }


    override fun onResume() {
        super.onResume()
        //添加自定义的拦截器
        Processor.addInterceptor(ReceivedServiceInterceptor())
        //执行
        Processor.newCall(Request("#https://abds"))
            .execute()
    }
}